repositories.bzl

```diff
@@ -0,0 +1,234 @@
+"""External dependencies for grpc-java."""
+
+def grpc_java_repositories(
+    omit_com_google_api_grpc_google_common_protos=False,
+    omit_com_google_code_findbugs_jsr305=False,
+    omit_com_google_code_gson=False,
+    omit_com_google_errorprone_error_prone_annotations=False,
+    omit_com_google_guava=False,
+    omit_com_google_instrumentation_api=False,
+    omit_com_google_protobuf=False,
+    omit_com_google_protobuf_java=False,
+    omit_com_google_protobuf_nano_protobuf_javanano=False,
+    omit_com_squareup_okhttp=False,
+    omit_com_squareup_okio=False,
+    omit_io_netty_buffer=False,
+    omit_io_netty_common=False,
+    omit_io_netty_transport=False,
+    omit_io_netty_codec=False,
+    omit_io_netty_codec_socks=False,
+    omit_io_netty_codec_http=False,
+    omit_io_netty_codec_http2=False,
+    omit_io_netty_handler=False,
+    omit_io_netty_handler_proxy=False,
+    omit_io_netty_resolver=False,
+    omit_io_netty_tcnative_boringssl_static=False):
+  """Imports dependencies for grpc-java."""
+  if not omit_com_google_api_grpc_google_common_protos:
+    com_google_api_grpc_google_common_protos()
+  if not omit_com_google_code_findbugs_jsr305:
+    com_google_code_findbugs_jsr305()
+  if not omit_com_google_code_gson:
+    com_google_code_gson()
+  if not omit_com_google_errorprone_error_prone_annotations:
+    com_google_errorprone_error_prone_annotations()
+  if not omit_com_google_guava:
+    com_google_guava()
+  if not omit_com_google_instrumentation_api:
+    com_google_instrumentation_api()
+  if not omit_com_google_protobuf:
+    com_google_protobuf()
+  if not omit_com_google_protobuf_java:
+    com_google_protobuf_java()
+  if not omit_com_google_protobuf_nano_protobuf_javanano:
+    com_google_protobuf_nano_protobuf_javanano()
+  if not omit_com_squareup_okhttp:
+    com_squareup_okhttp()
+  if not omit_com_squareup_okio:
+    com_squareup_okio()
+  if not omit_io_netty_buffer:
+    io_netty_buffer()
+  if not omit_io_netty_common:
+    io_netty_common()
+  if not omit_io_netty_transport:
+    io_netty_transport()
+  if not omit_io_netty_codec:
+    io_netty_codec()
+  if not omit_io_netty_codec_socks:
+    io_netty_codec_socks()
+  if not omit_io_netty_codec_http:
+    io_netty_codec_http()
+  if not omit_io_netty_codec_http2:
+    io_netty_codec_http2()
+  if not omit_io_netty_handler:
+    io_netty_handler()
+  if not omit_io_netty_handler_proxy:
+    io_netty_handler_proxy()
+  if not omit_io_netty_resolver:
+    io_netty_resolver()
+  if not omit_io_netty_tcnative_boringssl_static:
+    io_netty_tcnative_boringssl_static()
+  native.bind(
+    name = "guava",
+    actual = "@com_google_guava//jar",
+  )
+  native.bind(
+    name = "gson",
+    actual = "@com_google_code_gson//jar",
+  )
+
+def com_google_api_grpc_google_common_protos():
+  native.maven_jar(
+      name = "com_google_api_grpc_google_common_protos",
```

Comment:
please sort args, or at least make arg order consistent across this and the the rules below. (e.g. sha1 comes after name)